lib/shared/sdk/usbboot.js

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2017 resin.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * This work is heavily based on https://github.com/raspberrypi/usbboot
+ * Copyright 2016 Raspberry Pi Foundation
+ */
+
+'use strict'
+
+const _ = require('lodash')
+const Bluebird = require('bluebird')
+const usb = require('usb')
+
+/**
+ * @summary The number of USB interfaces in devices with a BCM2837 SoC
+ * @description
+ * Raspberry Pi 3, and CM3 use this SoC.
+ * @type {Number}
+ * @constant
+ */
+const INTERFACES_SOC_BCM2837 = 1
```

Comment:
Actually, this is `INTERFACES_SOC_BCM2835` - the [code comment](https://github.com/raspberrypi/usbboot/blob/master/main.c#L108) says "Handle 2837 where it can start with two interfaces", i.e. the 2837-handling is actually in the 'else' part of that code ;-)